pingcap/tidb
# *: avoid duplicated -engine=tiflash label

Signed-off-by: xhe <xw897002528@gmail.com>

### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary: Without this check, `-engine=tiflash` will be added again and again. It is a quick fix before the refactor of #22415 is merged. There is no way convenient way to test it.

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

```
#!/bin/sh

mycli -h 127.0.0.1 -P 4000 -u root --no-warn << "EOF"
set global tidb_enable_alter_placement=on;
drop database if exists test;
create database test;
EOF

mycli -h 127.0.0.1 -P 4000 -u root --no-warn << "EOF"
set global tidb_enable_alter_placement=on;
create table test.t1 (a int) partition by hash(a) partitions 2;
alter table test.t1 alter partition p0 alter placement policy role=leader;
alter table test.t1 alter partition p0 add placement policy role=voter replicas=1;
EOF

pd-ctl config placement-rules show
```

The result should not contain two `-engine=tiflash` labels.

### Release note 

- No release note
